### PR TITLE
MNT: Remove misc deprecated stuff in coordinates from a long time ago

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to address `coordinates` portion of things caught in #14664.

### CODE REVIEW

File: astropy/coordinates/solar_system.py
Comment (pllim) [inline astropy/coordinates/solar_system.py:556]: I didn't add this in change log because it is "hidden" anyway.

**Review by eerovaher — CHANGES_REQUESTED** [review]
`calc_moon()` has not been mentioned by name in the change log so far. It would be rather strange if the entry that announces its removal would be the first ever to name it, so I suggest removing the change log entry.

File: docs/changes/coordinates/14675.api.rst
Comment (pllim) [inline docs/changes/coordinates/14675.api.rst:1]: I didn't realize it wasn't mentioned before. Like this, @eerovaher ?

```suggestion
Removed deprecated angle utilities and orbital elements.
```

Comment (eerovaher) [reply docs/changes/coordinates/14675.api.rst:1]: I'd rather remove the change log entry entirely.

Comment (pllim) [reply docs/changes/coordinates/14675.api.rst:1]: To remove over 300 lines of code without any change log? I am not sure if I am comfortable with that. Are you sure absolutely no one is somehow still using this?

Comment (eerovaher) [reply docs/changes/coordinates/14675.api.rst:1]: Despite not having a name that begins with an underscore this function was considered to be private and was not documented, see https://docs.astropy.org/en/v4.3.1/search.html?q=calc_moon

Comment (pllim) [reply docs/changes/coordinates/14675.api.rst:1]: Still, the angle utilities being removed need change log?

Comment (pllim) [reply docs/changes/coordinates/14675.api.rst]: Accepted your suggestion. Thanks!

**Review by eerovaher — APPROVED** [review]
I'm leaving this open for now in case the other maintainers want to have a look, but I'll merge after a few days if they don't.

**Comment by eerovaher** [comment]
Thanks for the cleanup, @pllim

**Pull request merged**
